sdk/communication/azure-communication-phonenumbers/azure/communication/phonenumbers/siprouting/aio/_sip_routing_client_async.py

```diff
@@ -238,6 +241,30 @@ async def set_routes(
         ]
         await self._rest_service.sip_routing.update(body=SipConfiguration(routes=routes_internal), **kwargs)
 
+    @distributed_trace_async
+    async def test_routes_with_number_async(
+       self,
+        sip_configuration: Optional[Union[SipConfiguration, IO[bytes]]] = None,
+        *,
+        target_phone_number: str,
+        **kwargs: Any
+    ):
+        """Gets the list of routes matching the target phone number, ordered by priority.
+
+        Gets the list of routes matching the target phone number, ordered by priority.
+
+        :param sip_configuration: Sip configuration object to test with targetPhoneNumber. Is either a
+         SipConfiguration type or a IO[bytes] type. Default value is None.
+        :type sip_configuration: ~azure.communication.phonenumbers.siprouting.models.SipConfiguration
+         or IO[bytes]
+        :keyword target_phone_number: Phone number to test routing patterns against. Required.
+        :paramtype target_phone_number: str
+        :return: RoutesForNumber
```

Comment:
Please provide better description of returned object, something like:
List of routes matching the target number, provided in the same order of priority as in SipConfiguration.